File: code-samples/messaging/code-snippets/download-thread-msg-attachments.rb
require 'ringcentral'

#
# Read thread messages
#
def list_messages()
  begin
    query_params = {
      # threadStatus: "Open",
      # threadId: "...",
      # ownerExtensionIds: ["..."],
      # messageIds: ["..."],
      # creationTimeFrom: "...",
      # creationTimeTo: "...",
      perPage: 100
    }

    endpoint = "/restapi/v1.0/account/~/message-threads/messages"

    resp = $platform.get(endpoint, query_params)
    for msg in resp.body['records'] do
      # Parse and handle the message data

      # Check if this message has attachment(s)
      if msg["attachments"] && !msg["attachments"].empty?
        get_message_attachments(msg["attachments"])
      end
    end
  rescue StandardError => e
    puts e.message
  end
end

#
# Download attachment(s) and save to a local file
#
def get_message_attachments(attachments)
  begin
    for attachment in attachments
      fileName = attachment['filename']
      begin
        res = $platform.get(attachment['contentUri'])
        file = File.open(fileName, "w")
        file.write(res.body)
      rescue IOError => e
        puts e.getMessage()
      end
    end
  rescue StandardError => e
    puts (e)
  end
end

# Authenticate a user using a personal JWT token
def login()
  begin
    $platform.authorize( jwt: "RC_USER_JWT" )
    list_messages()
  rescue StandardError => e
    puts ("Unable to authenticate to platform. Check credentials." + e.to_s)
  end
end

File: code-samples/messaging/code-snippets/download-thread-msg-attachments.py
import json
from ringcentral import SDK

#
# Read thread messages
#
def list_messages():
    try:
        query_params = {
            # "threadStatus": "Open",
            # "threadId": "...",
            # "ownerExtensionIds": ["..."],
            # "messageIds": ["..."],
            # "creationTimeFrom": "...",
            # "creationTimeTo": "...",
            "perPage": 10
        }

        endpoint = "/restapi/v1.0/account/~/message-threads/messages"

        resp = platform.get(endpoint, query_params)
        jsonObj = resp.json_dict()
        for msg in jsonObj['records']:
            # Parse and handle the message data

            # Check if this message has attachment(s)
            if msg.get("attachments"):
                get_message_attachments(msg["attachments"])
    except Exception as e:
        print(str(e))
#
# Download attachment(s) and save to a local file
#
def get_message_attachments(attachments):
    for attachment in attachments:
        fileName = str(attachment["filename"])
        try:
            res = platform.get(attachment["contentUri"])
            file = open(("%s" % (fileName)),'wb')
            file.write(res.body())
            file.close()
        except ApiException as e:
            print (e.getMessage())

# Authenticate a user using a personal JWT token
def login():
    try:
      platform.login( jwt= "RC_USER_JWT" )
      list_messages()
    except Exception as e:
      print ("Unable to authenticate to platform. Check credentials." + str(e))
